### PR TITLE
21516 - EFT Over payment notification task

### PR DESCRIPTION
**Pull request by ochiu**
*Issue #:*
https://github.com/bcgov/entity/issues/21516

*Description of changes:*
- rename statement_due_task to eft_statement_due_task for clarity
- eft_overpayment_notification_task job for unlinked accounts with credits remaining for 30 days and linked over payments

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

File: jobs/payment-jobs/tests/jobs/test_eft_overpayment_notification_task.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    ('has-notifications', datetime(2024, 10, 1, 5),
+     datetime(2024, 10, 2, 5), []),
+    ('no-notifications', datetime(2024, 10, 1, 5),
+     datetime(2024, 10, 31, 5), None)
+])
+def test_over_payment_notification_unlinked(session, test_name, created_on_date, execution_date, assert_calls_override):
+    """Assert notification is being sent for unlinked accounts."""
+    unlinked_shortname, inactive_link_shortname, _ = create_unlinked_short_names_data(
+        created_on_date)
+    with patch('tasks.eft_overpayment_notification_task.send_email') as mock_mailer:
```
Comment (seeker25) [inline jobs/payment-jobs/tests/jobs/test_eft_overpayment_notification_task.py:123]: would recommend using testconf to mock these

File: jobs/payment-jobs/tasks/eft_overpayment_notification_task.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            cls.short_names = {}
+            if date_override:
+                cls.date_override = datetime.strptime(date_override, '%Y-%m-%d') if date_override else None
+                current_app.logger.info(f'Using date override : {date_override}')
+
+            # Get short names that have EFT credit rows created based on current / override date indicating payment
+            # was received and credits remaining indicate overpayment
+            linked_short_names = cls._get_today_overpaid_linked_short_names()
+
+            # Get short names that have credits remaining and have been unlinked for thirty days
```
Comment (seeker25) [inline jobs/payment-jobs/tasks/eft_overpayment_notification_task.py:100]: would bake this into the function name if possible

File: jobs/payment-jobs/tasks/eft_overpayment_notification_task.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        """Notify for over payments and short name unlinked for thirty days."""
+        try:
+            cls.short_names = {}
+            if date_override:
+                cls.date_override = datetime.strptime(date_override, '%Y-%m-%d') if date_override else None
+                current_app.logger.info(f'Using date override : {date_override}')
+
+            # Get short names that have EFT credit rows created based on current / override date indicating payment
+            # was received and credits remaining indicate overpayment
+            linked_short_names = cls._get_today_overpaid_linked_short_names()
```
Comment (seeker25) [inline jobs/payment-jobs/tasks/eft_overpayment_notification_task.py:98]: would bake this into the function name if possible

File: jobs/payment-jobs/tasks/eft_overpayment_notification_task.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+    Notify staff qualified receiver there is a pending unsettled amount on a short name due to overpayment or an account
+    has not been linked for thirty days.
+    """
+
+    date_override = None
+    short_names: dict = {}
+
+    @classmethod
+    def _get_base_query(cls):
```
Comment (seeker25) [inline jobs/payment-jobs/tasks/eft_overpayment_notification_task.py:42]: probably a better name for this? 

`get_shortnames_ids_with_credit_remaining` etc

File: jobs/payment-jobs/tasks/eft_overpayment_notification_task.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+            capture_message(
+                'Error on processing over payment notifications'
+                f'ERROR : {str(e)}', level='error')
+            current_app.logger.error(
+                'Error on processing over payment notifications', exc_info=True)
+
+    @classmethod
+    def _send_notifications(cls):
+        """Send over payment notification."""
+        if cls.short_names:
```
Comment (seeker25) [inline jobs/payment-jobs/tasks/eft_overpayment_notification_task.py:117]: save ya some indentation with `if cls.short_names is None: return`

**Comment by seeker25** [comment]
LGTM

**Review by seeker25 — APPROVED** [review]

**Review by seeker25 — APPROVED** [review]

**Comment by seeker25** [comment]
@ochiu probably need some cron job changes for these?

**Comment by ochiu** [comment]
> @ochiu probably need some cron job changes for these?

Yup, I will add the config

**Pull request merged**